init_project.sh

```diff
@@ -39,9 +39,31 @@ VPIPE_DIR=$(dirname "$0")
 
 cp -iv "$VPIPE_DIR/vpipe.config" "$PROJECT_DIR/"
 
+# guess activation command
+ACTIVATE=
+# search order:
+# - 'V-pipe' conda environment
+# - base conda environment
+for VPIPEENV in "$VPIPE_DIR/../miniconda3"{/envs/V-pipe,}; do
+	if [ -d "${VPIPEENV}" ] && [ -x "${VPIPEENV}/bin/activate" ] && [ -x "${VPIPEENV}/bin/conda" ] && [ -x "${VPIPEENV}/bin/snakemake" ]; then
+		echo "Conda environment found in ${VPIPEENV}"
+		if [ -x "${VPIPEENV}/envs/V-pipe/bin/snakemake" ]; then
+			ACTIVATE=". ${VPIPEENV}/bin/activate 'V-pipe'"
```

Comment:
Is this line ever reached?

It seems to me that in the first iteration, `VPIPENV = $VPIPE_DIR/../miniconda3"/envs/V-pipe` and thus the if-condition may be satisfied. This implies that `${VPIPEENV}/bin/snakemake` was found. However, this also implies that  `${VPIPEENV}/envs/V-pipe/bin/snakemake` would not be found. Thus, the script would activate the base environment and end the loop, which is probably not the intended behavior.

Reply:
Basically, three situations are considered:
 - ../miniconda3/envs/V-pipe/bin/activate 'base' (supports conda and snakemake having been installed together in the same sub environment, which is self-contained and could work as its own base)
 - ../miniconda3/bin/activate 'V-pipe' (V-pipe sub-environment cannot work as its own base) (this is what the sub condition is able to detect)
- ../miniconda3/bin/activate 'base' (everything installed normally)

3: is the normal way to do it.
1: is the fall-back mecanism happening at line 190.
2: can be implemented as a last-ditch effort if line 190 fails (but is not 100% guaranteed to work. Not doing that for the current release)

1 (and 2) happens when people have old versions of miniconda still hanging around and conflicting.

We *did* run into similar situation during the testing face. There's a reason why it's against SIB recommendations.